cmd/plural/crypto.go

```diff
@@ -180,13 +182,20 @@ func handleEncrypt(c *cli.Context) error {
 	if err != nil {
 		return err
 	}
-
-	prov, err := crypto.Build()
+	cryptoProv, err := crypto.Build()
+	if err != nil {
+		return err
+	}
+	prov, err := provider.GetProvider()
 	if err != nil {
 		return err
 	}
 
-	result, err := crypto.Encrypt(prov, data)
+	if prov.KeyID() != "" && cryptoProv.ID() != prov.KeyID() {
```

Comment:
we should probably also do this validation on `plural crypto unlock` too

Reply:
If there's a chicken-egg with the `workspace.yaml` being encrypted, then we should maybe create a new file in the repo for the key digest that is outside the encrypted list

Reply:
yes, good point